### PR TITLE
TestQrun is failing while running qrun command due to permission issue

### PR DESCRIPTION
**Pull request by Mahalty**
#### Bug/feature Description
* TestQrun is failing while running qrun command due to permission issue

#### Affected Platform(s)
* ALL

#### Cause / Analysis / Design
* In the test we are running job using qrun(using run_cmd function with as_script to True) which ends up asking for sudo password

#### Solution Description
* Instead of using run_cmd to run the job, use runjob server's API

#### Testing logs/output
* [TestQrun_afterfix.txt](https://github.com/PBSPro/pbspro/files/2608272/TestQrun_afterfix.txt)
* [TestQrun_bfr_fix.txt](https://github.com/PBSPro/pbspro/files/2608273/TestQrun_bfr_fix.txt)

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to this pull request and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [x] I have attached **test logs to this pull request** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: test/tests/functional/pbs_qrun.py [hunk 1]
```diff
@@ -57,20 +57,28 @@ def test_misuse_of_qrun(self):
...
+                     (self.mom.shortname, self.mom.shortname)
+        err_msg = 'qrun: Unknown node  "\'(%s)+(%s)\'"' % \
             (self.mom.shortname, self.mom.shortname)
-        self.assertIn(msg, ret['err'][-1])
+        try:
+            self.server.runjob(jobid=jid1, location=exec_vnode)
+        except PbsRunError as e:
+            self.assertIn(err_msg, e.msg[0])
+            self.logger.info('As expected qrun throws error: ' + err_msg)
+        else:
```
[the remark this answers is not in the document]
Comment (Mahalty) [reply test/tests/functional/pbs_qrun.py]: @anamikau: It runs when there is no exception

[the remark this answers is not in the document]
Comment (anamikau) [reply test/tests/functional/pbs_qrun.py]: ok thanks

**Comment by Mahalty** [comment]
Latest execution logs:
[Latest_Execution_logs_TestQrun.txt](https://github.com/PBSPro/pbspro/files/2614345/Latest_Execution_logs_TestQrun.txt)

**Comment by agrawalravi90** [comment]
Thanks for making the changes Zulekha, they look fine to me, once Anamika signs off again we can merge this.

**Comment by agrawalravi90** [comment]
@SudeshnaMoulik Please rebase

**Comment by Mahalty** [comment]
Thanks @anamikau and @agrawalravi90 for the review.
@agrawalravi90  I have squashed my commits and rebased my branch. Please take this PR forward

**Pull request merged**